addons/binding/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/handler/OpenUVHandler.java

```diff
@@ -0,0 +1,269 @@
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.openuv.internal.OpenUVConfiguration;
+import org.openhab.binding.openuv.json.OpenUVJsonResponse;
+import org.openhab.binding.openuv.json.OpenUVJsonResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializer;
+
+/**
+ * The {@link OpenUVHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+public class OpenUVHandler extends BaseThingHandler {
+    private Logger logger = LoggerFactory.getLogger(OpenUVHandler.class);
+    private static final int DEFAULT_REFRESH_PERIOD = 30;
+
+    OpenUVJsonResult openUVData;
+
+    private ScheduledFuture<?> refreshJob;
+    private ScheduledFuture<?> uvMaxJob;
+
+    private Gson gson;
+
+    public OpenUVHandler(Thing thing) {
+        super(thing);
+        gson = new GsonBuilder()
+                .registerTypeAdapter(DecimalType.class,
+                        (JsonDeserializer<DecimalType>) (json, type, jsonDeserializationContext) -> DecimalType
+                                .valueOf(json.getAsJsonPrimitive().getAsString()))
+                .registerTypeAdapter(ZonedDateTime.class, (JsonDeserializer<ZonedDateTime>) (json, type,
+                        jsonDeserializationContext) -> ZonedDateTime.parse(json.getAsJsonPrimitive().getAsString()))
+                .create();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing OpenUV handler.");
+
+        OpenUVConfiguration config = getConfigAs(OpenUVConfiguration.class);
+        String errorMsg = null;
+
+        if (StringUtils.trimToNull(config.apikey) == null) {
+            errorMsg = "Parameter 'apikey' is mandatory and must be configured";
+        }
+        if (config.refresh != null && config.refresh < 3) {
+            errorMsg = "Parameter 'refresh' must be at least 3 minutes to stay in free API plan";
+        }
+
+        if (errorMsg == null) {
+            updateStatus(ThingStatus.ONLINE);
+            startAutomaticRefresh();
+            startUVMaxScreening();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+        }
+    }
+
+    /**
+     * Start the job screening UV Max reached
+     */
+    private void startUVMaxScreening() {
+        if (uvMaxJob == null || uvMaxJob.isCancelled()) {
+            Runnable runnable = () -> {
+                try {
+                    if (openUVData != null) {
+                        long timeDiff = ChronoUnit.MINUTES.between(ZonedDateTime.now(ZoneId.systemDefault()),
+                                openUVData.getUVMaxTimeAsZDT());
+                        if (timeDiff == 0 || timeDiff == 1) {
+                            triggerChannel(UVMAXEVENT);
```

Comment:
Do I get it right, is this intended to trigger twice? Wouldn't it be better to schedule a job for exactly the uv max time (and adapt it whenever this changes, obviously) instead of testing every minute it is "there" already?